fs/rpmsgfs/rpmsgfs.c

```diff
@@ -252,6 +259,21 @@ static void rpmsgfs_mkpath(FAR struct rpmsgfs_mountpt_s  *fs,
     {
       strncat(path, &relpath[first], pathlen - strlen(path) - 1);
     }
+
+  while (fs->timeout > 0)
+    {
+      struct stat buf;
+      int ret;
+
+      ret = rpmsgfs_client_stat(fs->handle, fs->fs_root, &buf);
+      if (ret == 0)
+        {
+          break;
+        }
+
+      usleep(RPMSGFS_RETRY_DELAY_MS * USEC_PER_MSEC);
+      fs->timeout -= RPMSGFS_RETRY_DELAY_MS;
```

Comment:
This is not very good for delay calculation. I mean that the error is very big due to dependency on systick granularity. The better way is to use `clock_systime_timespec()`.

Reply:
We don't need so accurate time, just a timeout